pulser-core/pulser/noise_model.py

```diff
@@ -267,9 +267,10 @@ def to_tuple(obj: tuple) -> tuple:
                     "parameters. Defining only the relevant noise parameters "
                     "(without specifying the noise types) is recommended."
                 )
-            run_params_ = [
-                p for p in relevant_params if p in ("runs", "samples_per_run")
-            ]
+            # Only now that we know the relevant_params can we determine if
+            # we need to use the legacy defaults for the run parameters (ie in
+            # case they were not provided by the user)
+            run_params_ = relevant_params & {"runs", "samples_per_run"}
```

Comment:
We only use the legacy default values if the user specified `noise_types` explicitly (ie the default option). Otherwise, `runs` and `samples_per_run` are left as `None` when not specified and later raise an error upon validation

Reply:
Ah damn now I see it. Sorry !